test/ILLink.RoslynAnalyzer.Tests/TestChecker.cs

```diff
@@ -67,18 +73,39 @@ bool IsExpectedDiagnostic (AttributeSyntax attribute)
 			switch (attribute.Name.ToString ()) {
 			case "ExpectedWarning":
 				var args = TestCaseUtils.GetAttributeArguments (attribute);
-				if (args.TryGetValue ("ProducedBy", out var producedBy) &&
-					producedBy is MemberAccessExpressionSyntax memberAccessExpression &&
-					memberAccessExpression.Name is IdentifierNameSyntax identifierNameSyntax &&
-					identifierNameSyntax.Identifier.ValueText == "Trimmer")
-					return false;
+				if (args.TryGetValue ("ProducedBy", out var producedBy)) {
+					var producedByValues = GetProducedByValues (producedBy);
+					// Skip if this warning is not expected to be produced by any of the analyzers that we are currently testing.
+					return !producedByValues.Contains ("Analyzer") && (producedByValues.Intersect (TestingAnalyzers) == null);
```

Comment:
The condition looks inverted - I think it is skipping `ProducedBy.Analyzer` for example. I also am not sure an empty intersection will be null.